app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala

```diff
@@ -80,38 +78,76 @@ class DLCPaneModel(
+    } else {
+      InitMultiNonceOracleDialog.showAndWait(parentWindow.value)
+    }
 
     result match {
-      case Some((outcomes, contractInfo)) =>
+      case Some(contractInfo) =>
         val builder = new StringBuilder()
 
         val privKey = ECPrivateKey.freshPrivateKey
         val pubKey = privKey.schnorrPublicKey
-        val kValue = ECPrivateKey.freshPrivateKey
-        val rValue = kValue.schnorrNonce
-        val oracleInfo = SingleNonceOracleInfo(pubKey, rValue)
+        val (kValues, rValues, oracleInfo) = contractInfo match {
+          case SingleNonceContractInfo(_) =>
+            val kValue = ECPrivateKey.freshPrivateKey
+            val rValue = kValue.schnorrNonce
+            val oracleInfo = SingleNonceOracleInfo(pubKey, rValue)
+
+            (Vector(kValue), Vector(rValue), oracleInfo)
+          case MultiNonceContractInfo(_, _, numDigits, _) =>
+            val kValues =
+              0.until(numDigits).map(_ => ECPrivateKey.freshPrivateKey).toVector
+            val rValues = kValues.map(_.schnorrNonce)
+            val oracleInfo = MultiNonceOracleInfo(pubKey, rValues)
+
+            (kValues, rValues, oracleInfo)
+        }
 
         builder.append(
-          s"Oracle Public Key: ${pubKey.hex}\nEvent R value: ${rValue.hex}\n")
+          s"Oracle Public Key: ${pubKey.hex}\nEvent R values: ${rValues.map(_.hex).mkString(",")}\n")
         builder.append(s"Serialized Oracle Info: ${oracleInfo.hex}\n\n")
 
-        builder.append("Outcome hashes and amounts in order of entry:\n")
-        contractInfo.foreach {
-          case (str, amt) => builder.append(s"$str - ${amt.toLong}\n")
-        }
-        builder.append(s"\nSerialized Contract Info:\n${contractInfo.hex}\n\n")
-
-        builder.append("Outcomes and oracle sigs in order of entry:\n")
-        outcomes.zip(contractInfo.keys).foreach {
-          case (outcome, str) =>
-            val hash = str.serialized.head
-            val sig = privKey.schnorrSignWithNonce(hash, kValue)
-            builder.append(s"$outcome - ${sig.hex}\n")
+        builder.append(
+          s"\nSerialized Contract Info:\n${contractInfo.toTLV.hex}\n\n")
+
+        contractInfo match {
+          case contractInfo: SingleNonceContractInfo =>
+            builder.append("Outcomes and oracle sigs in order of entry:\n")
+            contractInfo.keys.foreach { outcome =>
+              val bytes = outcome.serialized.head
+              val hash = CryptoUtil.sha256(bytes).bytes
+              val sig = privKey.schnorrSignWithNonce(hash, kValues.head)
+              builder.append(s"$outcome - ${sig.hex}\n")
+            }
+          case contractInfo: MultiNonceContractInfo =>
+            builder.append("Oracle sigs:\n")
+            val max = UnsignedNumericOutcome(
+              contractInfo.outcomeVec.maxBy(_._2)._1)
+            val min = UnsignedNumericOutcome(
+              contractInfo.outcomeVec.minBy(_._2)._1)
```

Comment:
I think it would be nice to have something near the middle too, just so we can make sure to be testing 2 output CETs